core/storage/skill/gae_models.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for storing the skill data models."""
+
+from core.platform import models
+
+from google.appengine.ext import ndb
+
+(base_models,) = models.Registry.import_models([models.NAMES.base_model])
+
+
+class SkillSnapshotMetadataModel(base_models.BaseSnapshotMetadataModel):
+    """Storage model for the metadata for a skill snapshot."""
+    pass
+
+
+class SkillSnapshotContentModel(base_models.BaseSnapshotContentModel):
+    """Storage model for the content of a skill snapshot."""
+    pass
+
+
+class SkillModel(base_models.VersionedModel):
+    """Model for storing Skills.
+
+    This class should only be imported by the skill domain file, the
+    skill services file, and the skill model test file.
+    """
+    SNAPSHOT_METADATA_CLASS = SkillSnapshotMetadataModel
+    SNAPSHOT_CONTENT_CLASS = SkillSnapshotContentModel
+    ALLOW_REVERT = True
+
+    # The description of the skill.
+    description = ndb.StringProperty(required=True, indexed=True)
+    # A dict representing the skill contents.
+    skill_contents = ndb.JsonProperty(indexed=False)
```

Comment:
Is a schema version needed for this sub-dict?

Reply:
Yes, right? This is modelled by the StoryNode class, which has it own fields, which could change with time. So, for its migrations, I though schema version would be needed.

Reply:
I'm not sure you meant StoryNode class, but yeah, you're right. I think I might have misread: I meant to say "why isn't there a schema version" but on re-reading it looks like you have a field below for that, although it is mischaracterized as the schema version for the skill rather than the skill_contents dict.

Might want to rename it / adjust the docstring, and bring the two related fields more closely together.

Also, misconceptions appears to be a list of dicts. Maybe give more details on its structure in the comment. Also the Misconception object probably needs a schema version as well?

Reply:
Oh, I thought the file was story/gae_models.py, but yeah, I meant schema_version for skill_contents.
Ok, I'll add another schema_version for misconceptions. Also, I'll create a class Misconception in skill_domain.py, when I create the file. The structure would be specified there. So should I write it here also?

Reply:
If we have a Misconception domain object which tracks all this info then I think it's fine. Just make sure to say it's versioned and, in the docstring, point to the location which has info that allows the developer to understand what the contents are.